### ISSUE TITLE
Add support for arbitrary data mapping

### ISSUE THREAD

**Issue by Joshpolansky** [loupeteam/Loupe-UX#58]
add a data-map attribute that accepts json? Could also just see it accepting an array of key-value pairs. Could also allow selectors with the element as a scope.

something like:

OnUpdate(el){
 For (map [key, pvname ])
  El.querySelector(item) = machine.value(pvname)
}

or the simple version would just set like this:
![image](https://github.com/loupeteam/Loupe-UX/assets/669274/c06b3b9e-d1ac-4640-aa6f-8bb509bc5956)

**Comment by Joshpolansky** [comment]
I might also propose a more complex mapping, to accommodate all the other possibilities.

"[target attr]":{
"Data-var-name":"[pvname]",
"SourceUnits":"[unit type]",
"DisplayUnits": "[unit type]",
..
}

**Comment by sclaiborne** [comment]
To clarify in the default case you are asking for data-map to map value (pv) to key (attribute) on the same element. So...
```html
<ob-rudder-medium rudderAngle="0"
   rudderSetPointAngle="0"
   showPortStarboard="false"
   showSetPoint="false"
   data-map="{ 'rudderAngle' : 'tmplitTest:tmplit.Slider' , ' rudderSetPointAngle' : 'tmp1itTest:tmplit.Numericlnput' }"/>
```
Assuming variables tmplitTest:tmplit.Slider = 100 and tmp1itTest:tmplit.Numericlnput  = 200 Then
```html
<ob-rudder-medium rudderAngle="100"
   rudderSetPointAngle="200"
   showPortStarboard="false"
   showSetPoint="false"
   data-map="{ 'rudderAngle' : 'tmplitTest:tmplit.Slider' , ' rudderSetPointAngle' : 'tmp1itTest:tmplit.Numericlnput' }"/>
```

Then if ` 'rudderAngle' :` started with a `{`, you would parse it for additional information?

**Comment by Joshpolansky** [comment]
Yes, but actually.. I would parse it and check if the parsed value was a string or object. If it's a string we assume it's a PV, we f it's an object we get the parameters from the object
